@xen-orchestra/xapi/network.mjs

```diff
@@ -0,0 +1,14 @@
+export default class Network {
+  async setMtu(network, mtu) {
+    await network.set_MTU(mtu)
+
+    // The MTU will not be updated for VIFs of paused VM, but we can't unplug/replug VIF on a paused VM
+    const pluggedVIFs = network.$VIFs.filter(vif => vif.currently_attached && vif.$VM.power_state === 'Running')
+    const pluggedPIFs = network.$PIFs.filter(pif => pif.currently_attached)
+
+    await Promise.allSettled(pluggedVIFs.map(vif => this.callAsync('VIF.unplug', vif.$ref)))
+    await Promise.allSettled(pluggedPIFs.map(pif => this.callAsync('PIF.plug', pif.$ref)))
+    await Promise.allSettled(pluggedPIFs.map(pif => this.callAsync('PIF.plug', pif.$ref)))
+    await Promise.allSettled(pluggedVIFs.map(vif => this.callAsync('VIF.plug', vif.$ref)))
```

Comment:
I think that , in case of error, we should try to plug back what we unplug , I would use $defer on this.

For example , what do you think of something like this : 
```js
await this.callAsync('VIF.unplug', vif.$ref)
$defer(()=>this.callAsync('VIF.plug', vif.$ref))
```
@julien-f are the deferred executed in the right order ( the laste deferred added should be executed first) ?

Reply:
> Note: the deferreds are executed in the reverse order of their declarations.
>
> -- https://github.com/JsCommunity/golike-defer